### DISCUSSION TITLE
Bot crashing after few days online

### DISCUSSION THREAD

**Discussion by iknowbr12** [Rapptz/discord.py#7105] [Q&A]
Hi! 

My bot after few days ends crashing (very delay/lag, high ping), with the error:
![Erro_Lord](https://user-images.githubusercontent.com/40132324/123295495-24d15100-d4ec-11eb-8973-395d9c67ea4d.png)

I don't believe it's exactly my server (4gb ram, 2v core), it usually happens after 3 consecutive days on.
It's worth saying that during this error, I notice that my processor doesn't work well, the bot process has high usage, while the server is at half of usage:
![Screenshot_1](https://user-images.githubusercontent.com/40132324/123295894-88f41500-d4ec-11eb-9f9e-80aa0e15e439.png)
Currently, I solve this problem by restarting everything, but it's not correct, I don't think that should happen.

I suspect:
- Server network connection issues
- Some system overloads, something like an infinite loop after a few days;
- Some system does not deal well with network outages or oscillations and ends up overloading the server.

Any help is welcome, how can I identify this error? The error just tells me that it occurs in a random ctx.send, it doesn't make sense to me.

**Comment by AbstractUmbra** [comment]
The broken pipe error implies a small network blackout to me. Do you have network errors on this host?

**Reply by iknowbr12** [reply]
I don't see any errors in the server connection, after the bug occurs I do network tests (speedtest) and I don't see any problems (speed seems good and stable).
I using GalaxyGate server.

**Reply by AbstractUmbra** [reply]
From the top of my head, GalaxyGate is known for rather poor infrastructure.

This broken pipe error was rather common in the help channels in the official server for this library. It does strongly suggest a network blip.
You can see the traceback begins when your bot makes a web request with a message send operation.

**Reply by circuitsacul** [reply]
I use GalaxyGate for a fairly large bot and it performs very well even when running for weeks, so I don't think this is purely a GalaxyGate issue.
